jobs/MeterInstanceJob.js

```diff
@@ -0,0 +1,186 @@
+    });
+  }
+
+  static getInstanceEvents() {
+    const options = {
+      resourceGroup: CONST.APISERVER.RESOURCE_GROUPS.INSTANCE,
+      resourceType: CONST.APISERVER.RESOURCE_TYPES.SFEVENT,
+      query: {
+        labelSelector: `meter_state in (${CONST.METER_STATE.TO_BE_METERED},${CONST.METER_STATE.FAILED})`
+      }
+    };
+    return apiServerClient.getResources(options);
+  }
+
+  static meter(events) {
+    logger.info(`Number of events to be metered in this run - ${events.length}`);
+    return Promise.try(() => {
+      let successCount = 0,
+        failureCount = 0,
+        failedEvents = [];
+      return Promise.map(events, (event) => {
+          return this
+            .sendEvent(event)
+            .then((status) => {
+              if (status) {
+                successCount++;
+              } else {
+                failureCount++;
+                failedEvents.push(event);
+              }
+            })
+            .catch(err => {
+              logger.error(`Error occurred while metering event : `, err);
+            });
+        }).then(() => {
+          return {
+            totalEvents: events.length,
+            success: successCount,
+            failed: failureCount,
+            failedEvents: failedEvents
+          };
+        })
+        .catch(err => {
+          logger.error(`Error occurred while metering all events : `, err);
+        });
+
+    });
+  }
+
+  static isServicePlanExcluded(options) {
+    options = JSON.parse(options);
+    const serviceId = _.get(options, 'service.id');
+    const planId = _.get(options, 'service.plan');
+    logger.info(`Checking if service ${serviceId}, plan: ${planId} is excluded`);
+    const serviceName = this.getServiceNameFromServiceGUID(serviceId);
+    const planName = this.getPlanSKUFromPlanGUID(serviceId, planId);
+    const excluded_service_names = _.map(config.metering.excluded_service_plans, p => p.service_name);
+    if (_.indexOf(excluded_service_names, serviceName) >= 0) {
+      const excluded_plans = _
+        .chain(config.metering.excluded_service_plans)
+        .filter(p => p.service_name === serviceName)
+        .head()
+        .get('plan_sku_names')
+        .value();
+      logger.info(`Excluded plans for ${serviceName}:`, excluded_plans);
+      if (_.indexOf(excluded_plans, planName) >= 0) {
+        logger.info(`The metering event guid: ${options.id} with ${serviceName}, ${planName} is excluded`);
+        return true;
+      }
+    }
+    return false;
+  }
+
+  static enrichEvent(options) {
+    // Add region , service name and plan sku name of the event
+    options = JSON.parse(options);
+    const serviceId = _.get(options, 'service.id');
+    const planId = _.get(options, 'service.plan');
+    logger.info(`Enriching the metering event ${serviceId}, plan: ${planId}`);
+    options.service.id = this.getServiceNameFromServiceGUID(serviceId);
+    options.service.plan = this.getPlanSKUFromPlanGUID(serviceId, planId);
```

Comment:
You are getting `planId` from `options.service.plan` and then storing sku in it. Does not seem right. Can you not have an attribute like `option.service.plan.sku` ?

Reply:
options are populated in webhook. 
Webhook has no knowledge of catalog, but only of the current CRD on which it is operating. 
So we store the plan guid and change it to sku before sending.

Reply:
Done !!